### PR TITLE
Update Developer Guide on new NFRs and Glossary

### PR DESCRIPTION
**Pull request by yuhinarita**

### CODE REVIEW

**Review by Ryan-Reno — COMMENTED** [review]
LGTM

**Review by Ryan-Reno — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -329,16 +329,22 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-*{More to be added}*
+1. **Environment**: Should work on any _mainstream_ OS as long as it has Java `11` or above installed.
+2. **Performance**: Should respond to user interaction within 3 seconds at most for typical usage (unless it is lagging due to reasons external to the app).
+3. **Performance**: Should be able to hold up to 1000 _coursemates_ without a noticeable sluggishness in performance (as specified above) for typical usage.
+4. **Resilience**: Should gracefully handle commonly anticipated errors (e.g. incorrect _command_ input) without crashing or losing saved data.
+5. **Accessibility**: Should notify the user whether a _command_ is successful or has failed.
+6. **Accessibility**: Should be accessible to English speakers with average typing speed.
+7. **Accessibility**: Usage of basic _commands_ (e.g. add, edit) should be learnable within a day.
```
Comment (benson1029) [inline docs/DeveloperGuide.md:339]: Can consider adding: Data in the app should be persistently stored.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -329,16 +329,22 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-*{More to be added}*
+1. **Environment**: Should work on any _mainstream_ OS as long as it has Java `11` or above installed.
+2. **Performance**: Should respond to user interaction within 3 seconds at most for typical usage (unless it is lagging due to reasons external to the app).
+3. **Performance**: Should be able to hold up to 1000 _coursemates_ without a noticeable sluggishness in performance (as specified above) for typical usage.
+4. **Resilience**: Should gracefully handle commonly anticipated errors (e.g. incorrect _command_ input) without crashing or losing saved data.
+5. **Accessibility**: Should notify the user whether a _command_ is successful or has failed.
```
Comment (benson1029) [inline docs/DeveloperGuide.md:336]: Doesn't seem very useful to me.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -329,16 +329,22 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+6. **Accessibility**: Should be accessible to English speakers with average typing speed.
```
Comment (benson1029) [inline docs/DeveloperGuide.md:337]: I think we should keep "faster than using the mouse" since CLI-preferred is the course-wide requirement.

**Review by benson1029 — APPROVED** [review]
LGTM. Minor things on the NFRs can be fixed later

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (yuhinarita) [reply docs/DeveloperGuide.md:339]: Sure

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (yuhinarita) [reply docs/DeveloperGuide.md:336]: got it. will fix later in another pr

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (yuhinarita) [reply docs/DeveloperGuide.md:337]: got it. will fix later in another pr

**Pull request merged**